cpp/tests/c_api/legacy_k_truss_test.c

```diff
@@ -0,0 +1,243 @@
+
+    size_t num_result_edges   = cugraph_type_erased_device_array_view_size(src);
+    size_t num_result_offsets = cugraph_type_erased_device_array_view_size(offsets);
+
+    vertex_t h_result_src[num_result_edges];
+    vertex_t h_result_dst[num_result_edges];
+    weight_t h_result_wgt[num_result_edges];
+    size_t h_result_offsets[num_result_offsets];
+
+    ret_code = cugraph_type_erased_device_array_view_copy_to_host(
+      resource_handle, (byte_t*)h_result_src, src, &ret_error);
+    TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "copy_to_host failed.");
+
+    ret_code = cugraph_type_erased_device_array_view_copy_to_host(
+      resource_handle, (byte_t*)h_result_dst, dst, &ret_error);
+    TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "copy_to_host failed.");
+
+    if (wgt != NULL){
+      ret_code = cugraph_type_erased_device_array_view_copy_to_host(
+        resource_handle, (byte_t*)h_result_wgt, wgt, &ret_error);
+      TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "copy_to_host failed.");
+    }
+
+
+    ret_code = cugraph_type_erased_device_array_view_copy_to_host(
+      resource_handle, (byte_t*)h_result_offsets, offsets, &ret_error);
+    TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "copy_to_host failed.");
+
+    TEST_ASSERT(test_ret_value, num_result_edges == num_expected_edges, "results not the same size");
+
+    for (size_t i = 0; (i < num_expected_offsets) && (test_ret_value == 0); ++i) {
+      TEST_ASSERT(test_ret_value,
+                  h_expected_offsets[i] == h_result_offsets[i],
+                  "graph offsets should match");
+    }
+
+  for (size_t i = 0; (i < num_expected_edges) && (test_ret_value == 0); ++i) {
+    bool_t found = FALSE;
+    for (size_t j = 0; (j < num_expected_edges) && !found; ++j) {
+      if ((h_expected_src[i] == h_result_src[j]) && (h_expected_dst[i] == h_result_dst[j]))
+          if (wgt != NULL){
+            found = (nearlyEqual(h_expected_wgt[i], h_result_wgt[j], 0.001));
+          }
+          else{
+            found = TRUE;
+          }
+    }
+    TEST_ASSERT(test_ret_value, found, "extracted an edge that doesn't match");
+  }
+
+#if 0
+    ret_code = cugraph_type_erased_device_array_view_copy_to_host(
+      resource_handle, (byte_t*)h_result_wgt, wgt, &ret_error);
+    TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "copy_to_host failed.");
+#endif
+
+
+    cugraph_type_erased_device_array_view_free(src);
+    cugraph_type_erased_device_array_view_free(dst);
+    cugraph_type_erased_device_array_view_free(wgt);
+    cugraph_type_erased_device_array_view_free(offsets);
+    cugraph_induced_subgraph_result_free(result);
+  }
+
+  cugraph_sg_graph_free(graph);
+  cugraph_free_resource_handle(resource_handle);
+  cugraph_error_free(ret_error);
+
+  return test_ret_value;
+}
+
+int test_k_truss()
+{
+  size_t num_edges    = 16;
+  size_t num_vertices = 6;
+  size_t k = 2;
```

Comment:
k = 2 is not very interesting, just returns the whole graph.  Perhaps try k = 3?

Reply:
I believe it crashes but let me double check. Maybe I should get a more interesting datasets if it is the case?

Reply:
Never mind. it works for 3 and we actually get a subgraph smaller than the initital graph. Let me update this tests accordingly